test/gtest/uct/ib/test_dc.cc

```diff
@@ -496,9 +496,28 @@ class test_dc_flow_control : public test_rc_flow_control {
         }
     }
 
-    virtual void test_pending_grant(int wnd, uint64_t *wait_fc_seq = NULL)
+    virtual void wait_fc_hard_resend(entity *e)
     {
-        test_rc_flow_control::test_pending_grant(wnd, wait_fc_seq);
+        uct_dc_mlx5_iface_t *iface = ucs_derived_of(e->iface(),
+                                                    uct_dc_mlx5_iface_t);
+        uct_dc_mlx5_ep_t *ep       = ucs_derived_of(e->ep(0),
+                                                    uct_dc_mlx5_ep_t);
+
+        khiter_t it = kh_get(uct_dc_mlx5_fc_hash, &iface->tx.fc_hash,
+                             (uint64_t)ep);
+        ASSERT_NE(it, kh_end(&iface->tx.fc_hash));
+
+        uct_dc_mlx5_ep_fc_entry_t *fc_entry = &kh_value(&iface->tx.fc_hash,
+                                                        it);
+        ucs_time_t fc_send_time = fc_entry->send_time;
+        wait_for_value_change(&fc_entry->send_time, true);
```

Comment:
could fc_entry be removed?

Reply:
no, it shouldn't be empty, since we shouldn't receive FC_PURE_GRANT from a peer.
but in order to make sure that FC_PURE_GRANT won't be received, I've added progressing only sender entity